modin/pandas/dataframe.py

```diff
@@ -2550,7 +2550,9 @@ def setitem_unhashable_key(df, value):
                 value = value.T.reshape(-1)
                 if len(self) > 0:
                     value = value[: len(self)]
-            if not isinstance(value, Series):
+            if not isinstance(
+                value, (Series, pandas.core.arrays.categorical.Categorical)
```

Comment:
```suggestion
                value, (Series, pd.Categorical)
```
let's be consistent and use our own dtype